### PR TITLE
Upgrade our argo fork to the latest version

### PR DESCRIPTION
**Pull request by decarboxy**
This PR consists of a merge of `argoproj/argo:master` into our fork, as well as a merge of https://github.com/argoproj/argo/pull/1067.  We're pulling in about 8 months of bugfixes here. 

based on the changes to the make file im guessing our jenkins pipeline will need to change a lot

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,94 @@
...
+- Pod creation with secret volumemount (#1318)
+- Export the methods of `KubernetesClientInterface` (#1294)
+
+
+## 2.3.0-rc1 (2019-04-10)
+
+### Notes about upgrading from v2.2
+
+* Secrets are passed to the wait sidecar using volumeMounts instead of performing K8s API calls
+  performed by the. This is much more secure since it limits the privileges of the workflow pod
```
Comment (squidlegs) [inline CHANGELOG.md:19]: This isn't even.

Do we have to do something here?

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,5 +1,94 @@
...
+  to no longer require namespace level secret access. However, as a consequence, workflow pods which
+  reference a secret that does not exist, will now indefinitely stay in a Pending state, as opposed
+  to the previous behavior of failing during runtime.
+
+
+### Deprecation Notice
+The workflow-controller-configmap introduces a new config field, `executor`, which is a container 
+spec and provides controls over the executor sidecar container (i.e. `init`/`wait`). The fields
+`executorImage`, `executorResources`, and `executorImagePullPolicy` are deprecated and will be
```
Comment (squidlegs) [inline CHANGELOG.md:28]: 😭

File: ROADMAP.md [hunk 3]
```diff
@@ -1,14 +1,13 @@
...
-The following are candidate items for v2.2 release
-
-* Workflow composability - support for Jsonnet in CLI
-* Queuing / Admission control - ability to limit number of concurrent workflows
-* Scheduling - investigate k8s PriorityClasses and re-use in workflows
-* Persistence - workflow history/state
-* `argo run` to run workflows against clusters without a controller - #794
-* UI – filtering to improve performance
+## v2.4
+* Persistence - support offloading of workflow state into database layer
```
Comment (squidlegs) [inline ROADMAP.md:4]: image

File: ROADMAP.md [hunk 4]
```diff
@@ -1,14 +1,13 @@
...
+* Large workflow support (enabled by persistence feature)
```
Comment (squidlegs) [inline ROADMAP.md:5]: *more heavy breathing*

File: workflow/validate/lint.go [hunk 5]
```diff
@@ -6,6 +6,7 @@ import (
 	"path/filepath"
 
 	"github.com/argoproj/pkg/json"
```
Comment (squidlegs) [inline workflow/validate/lint.go:8]: Are we going to need this guy too?

Comment (decarboxy) [reply workflow/validate/lint.go:8]: not unless we need to modify it https://github.com/argoproj/pkg

**Review by squidlegs — APPROVED** [review]

**Pull request merged**